### PR TITLE
Enhance ppa tests

### PR DESCRIPTION
**Pull request by EugenM-SAP**

### CODE REVIEW

**Review by sorin-iovita — APPROVED** [review]

File: common/persistence/src/test/java/app/coronawarn/datadonation/common/persistence/service/PpaDataServiceTest.java [hunk 1]
```diff
@@ -1,15 +1,20 @@
 package app.coronawarn.datadonation.common.persistence.service;
 
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
-import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.mock;import java.time.LocalDate;
```
Comment (hilmarf) [inline common/persistence/src/test/java/app/coronawarn/datadonation/common/persistence/service/PpaDataServiceTest.java:4]:
```suggestion
import static org.mockito.Mockito.mock;

import java.time.LocalDate;
```

File: services/ppac/src/test/java/app/coronawarn/datadonation/services/ppac/android/testutils/AndroidTestUtils.java [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package app.coronawarn.datadonation.services.ppac.android.testutils;
+
+public class AndroidTestUtils {
+
+}
```
Comment (hilmarf) [inline services/ppac/src/test/java/app/coronawarn/datadonation/services/ppac/android/testutils/AndroidTestUtils.java:5]: what's this good for?

**Review by hilmarf — APPROVED** [review]

File: services/ppac/src/test/java/app/coronawarn/datadonation/services/ppac/android/controller/AndroidControllerTest.java [hunk 3]
```diff
@@ -65,209 +75,380 @@
...
     @Test
-    @Disabled("Temporarily disabled until test refactoring commited")
     void checkResponseStatusForValidNonce() throws IOException {
-      ppacConfiguration.getAndroid().setCertificateHostname("localhost");
-      ResponseEntity<Void> actResponse = executor.executePost(buildPayloadWithValidNonce());
+      ppacConfiguration.getAndroid().setDisableNonceCheck(false);
+      ResponseEntity<DataSubmissionResponse> actResponse = executor.executePost(buildPayloadWithValidNonce());
       assertThat(actResponse.getStatusCode()).isEqualTo(NO_CONTENT);
+      assertDataWasSaved();
     }
```
Comment (hilmarf) [inline services/ppac/src/test/java/app/coronawarn/datadonation/services/ppac/android/controller/AndroidControllerTest.java:130]: @EugenM-SAP can you please check why this test doesn't pass in CircleCi build?

**Review by hilmarf — CHANGES_REQUESTED** [review]
checkResponseStatusForValidNonce is failing

**Review by hilmarf — APPROVED** [review]

**Pull request merged**